metadata-jobs/mae-consumer/src/main/java/com/linkedin/metadata/kafka/hook/UpdateIndicesHook.java

```diff
@@ -109,72 +108,110 @@ public UpdateIndicesHook(
   }
 
   @Override
-  public void invoke(@Nonnull MetadataChangeLog event) {
-    EntitySpec entitySpec;
-    try {
-      entitySpec = _entityRegistry.getEntitySpec(event.getEntityType());
-    } catch (IllegalArgumentException e) {
-      log.error("Error while processing entity type {}: {}", event.getEntityType(), e.toString());
+  public void invoke(@Nonnull final MetadataChangeLog event) {
+    if (UPDATE_CHANGE_TYPES.contains(event.getChangeType())) {
+      handleUpdateChangeEvent(event);
+    } else if (event.getChangeType() == ChangeType.DELETE) {
+      handleDeleteChangeEvent(event);
+    }
+  }
+
+  /**
+   * This very important method processes {@link MetadataChangeLog} events
+   * that represent changes to the Metadata Graph.
+   *
+   * In particular, it handles updating the Search, Graph, Timeseries, and
+   * System Metadata stores in response to a given change type to reflect
+   * the changes present in the new aspect.
+   *
+   * @param event the change event to be processed.
+   */
+  private void handleUpdateChangeEvent(@Nonnull final MetadataChangeLog event) {
+
+    final EntitySpec entitySpec = getEventEntitySpec(event);
+    final Urn urn = EntityKeyUtils.getUrnFromLog(event, entitySpec.getKeyAspectSpec());
+
+    if (!event.hasAspectName() || !event.hasAspect()) {
+      log.error("Aspect or aspect name is missing. Skipping aspect processing...");
       return;
     }
-    Urn urn = EntityKeyUtils.getUrnFromLog(event, entitySpec.getKeyAspectSpec());
 
-    if (VALID_CHANGE_TYPES.contains(event.getChangeType())) {
+    AspectSpec aspectSpec = entitySpec.getAspectSpec(event.getAspectName());
+    if (aspectSpec == null) {
+      throw new RuntimeException(
+          String.format("Failed to retrieve Aspect Spec for entity with name %s, aspect with name %s. Cannot update indices for MCL.",
+              event.getEntityType(),
+              event.getAspectName()));
+    }
 
-      if (!event.hasAspectName() || !event.hasAspect()) {
-        log.error("Aspect or aspect name is missing");
-        return;
-      }
+    RecordTemplate aspect = GenericRecordUtils.deserializeAspect(
+        event.getAspect().getValue(),
+        event.getAspect().getContentType(),
+        aspectSpec);
+    GenericAspect previousAspectValue = event.getPreviousAspectValue();
+    RecordTemplate previousAspect = previousAspectValue != null
+        ? GenericRecordUtils.deserializeAspect(previousAspectValue.getValue(), previousAspectValue.getContentType(), aspectSpec)
+        : null;
+
+    // Step 0. If the aspect is timeseries, add to its timeseries index.
+    if (aspectSpec.isTimeseries()) {
+      updateTimeseriesFields(event.getEntityType(), event.getAspectName(), urn, aspect, aspectSpec,
```

Comment:
Here's the important part: Only update timeseries index if its a time series aspect. Do everything else either way.